### PR TITLE
Create tune.ipynb

### PR DESCRIPTION
**Pull request by torrwill**
# Model Tuner (WIP)

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Documentation update
- [x] Refactoring
- [ ] Hotfix

## Description
The ModelTuner class applies Bayesian optimization and evaluates them using different scoring metrics for classification and regression tasks. It includes methods for tuning individual models, evaluating them on test data, storing the results, and visualizing performance metrics such as confusion matrices or regression plots.

## How to Run:
```py
X_train_clf, X_test_clf, y_train_clf, y_test_clf = train_test_split(digits.data, digits.target, test_size=0.3, random_state=42)
X_train_reg, X_test_reg, y_train_reg, y_test_reg = train_test_split(modelData, cleanTarget, test_size=0.3, random_state=42)

tuner = ModelTuner(models_param_spaces, scoring_criteria)

tuner.tune_all_models('classification', 'bayesian', X_train_clf, y_train_clf, X_test_clf, y_test_clf)
tuner.tune_all_models('regression', 'bayesian', X_train_clf, y_train_clf, X_test_clf, y_test_clf)
```

## Additional Information
image
image
image
image

The metrics plots will probably be modified to subplot based on scoring metric and model. Open to suggestions!
Note: feature importance still needs a plot.

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by WatsonWBlair — APPROVED** [review]
lgtm
